darshan-util/pydarshan/darshan/experimental/plots/plot_io_cost.py

```diff
@@ -39,14 +39,47 @@ def get_by_avg_series(df: Any, mod_key: str, nprocs: int) -> Any:
         f"{mod_key}_F_WRITE_TIME",
         f"{mod_key}_F_META_TIME",
     ]
-    df = df.filter(cols, axis=1)
+    by_avg_series = df.filter(cols, axis=1).sum(axis=0) / nprocs
+    # reindex to ensure 3 rows are always created
+    by_avg_series = by_avg_series.reindex(cols, fill_value=0.0)
     # rename the columns so the labels are automatically generated when plotting
     name_dict = {cols[0]: "Read", cols[1]: "Write", cols[2]: "Meta"}
-    df.rename(columns=name_dict, inplace=True)
-    by_avg_series = df.sum(axis=0) / nprocs
+    by_avg_series.rename(index=name_dict, inplace=True)
     return by_avg_series
 
 
+def combine_hdf5_modules(df: Any) -> Any:
+    """
+    Combines the "H5F" and "H5D" rows in the input dataframe into
+    a single entry under the "HDF5" title.
+
+    Parameters
+    ----------
+    df: a ``pd.DataFrame`` containing the average read, write, and meta
+    times for various pydarshan modules (i.e. "POSIX", "MPI-IO", "STDIO").
+
+    Returns
+    -------
+    Modified version of the input dataframe, where
+    if either or both "H5F" and "H5D" modules are
+    present, they have been renamed and/or summed
+    under a new index "HDF5", if available.
+
+    Notes
+    -----
+    If a single HDF5-related module is present it will
+    be renamed as "HDF5". If no HDF5-related modules
+    are present the dataframe will be unchanged.
+
+    """
+    # replace the H5D/H5F indexes with HDF5 and sum them
+    df = df.reset_index().replace(to_replace=r"H5\D", value="HDF5", regex=True)
```

Comment:
Do we really want a regex that matches `H5` + any non-digit character? There are other HDF5 modules like `H5Z`, `H5A` and so on, which would also match that.

While I'm not aware of any plans by the `darshan` team to expand to those, it strikes me as slightly safer not to silently condense pretty much any HDF5 module, and instead explicitly condense the two modules you want: i.e., something like `r"H5[FD]"`

Reply:
We have to explicitly add modules upstream for them to make it this far in the code, so other HDF5 modules would never appear here even if they were present in the report. But I like the more-explicit approach you've suggested, so I'll go ahead and change that. Thanks for the suggestion